### PR TITLE
Update to OCaml 4.04.2

### PR DESCRIPTION
**Pull request by gaborigloi**
I've tested the new `Vagrantfile` and it worked.

Signed-off-by: Gabor Igloi <gabor.igloi@citrix.com>

### CODE REVIEW

File: Vagrantfile [hunk 1]
```diff
@@ -46,16 +46,18 @@ Vagrant.configure("2") do |config|
...
-    opam depext -y xapi
+    opam depext -y xapi conduit.0.14.5
+    opam install -y conduit.0.14.5
     opam install --deps-only xapi
 
 # Verify xapi can be built
     sudo apt-get install -y git
     git clone git://github.com/xapi-project/xen-api
+    # To avoid stack overflow error:
+    ulimit -s unlimited
```
Comment (mseri) [inline Vagrantfile]: Please use `ulimit -s 16384` instead of allowing unlimited stack

File: Vagrantfile [hunk 2]
```diff
@@ -46,16 +46,18 @@ Vagrant.configure("2") do |config|
...
 # Install opam and dependencies for compiling xapi
     sudo apt-get install -y opam m4 libxen-dev
-    opam init -a --compiler=4.02.3 -y
+    opam init -a --compiler=4.04.2 -y
     eval `opam config env`
     opam remote add xs-opam git://github.com/xapi-project/xs-opam
-    opam install -y cppo_ocamlbuild
-    opam depext -y xapi
+    opam depext -y xapi conduit.0.14.5
+    opam install -y conduit.0.14.5
```
Comment (Frezzle) [inline Vagrantfile:53]: Should this be fixed by xs-opam?

Comment (gaborigloi) [reply Vagrantfile:53]: Yes, I'm not sure atm where to add the constraint.

Comment (mseri) [reply Vagrantfile:53]: Unfortunately it seems that the main opam overrides xs-opam, so unless we remove the default repository we need that pin

Comment (mseri) [reply Vagrantfile:53]: Otherwise the issue would have been fixed by https://github.com/xapi-project/xs-opam/pull/135

Comment (gaborigloi) [reply Vagrantfile:53]: So should we fix it in the main repo as well?

Comment (mseri) [reply Vagrantfile:53]: Why not?

Comment (gaborigloi) [reply Vagrantfile:53]: There is actually already an upper bound on `conduit` for `cohttp` in the main repo 🤔  :
```
"conduit" {>= "0.14.0" & <"0.16.0"}
```

Comment (gaborigloi) [reply Vagrantfile:53]: I think we can merge this nevertheless.

**Review by mseri — APPROVED** [review]

**Pull request merged**